### ISSUE TITLE
Is there a eval() function available in yolo_v8 as is it available in keras application models?

### ISSUE THREAD

**Issue by mimansa1907** [ultralytics/ultralytics#6236]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

In order to get a performance summary of the validation folder we use model.val(data=_path_to_yaml_), is a there a similar function available to get performance summary of test folder? My test folder also has labels so I can perform an evaluation operation there to check model performance on unseen data.

**Comment by glenn-jocher** [comment]
@mimansa1907 yes, YOLOv8 provides functionality for evaluating model performance on a separate test dataset which has associated labels. You can use the `val` mode with a path to a data configuration YAML file specifying your test dataset path and labels. The process is similar to the evaluation of the validation dataset, except you will point it to your test set.

Inside your YAML file, typically you define `train`, `val`, and `test` paths. If you want to evaluate on the test set, make sure the `test` path points to your test folder. Then you run the `val` mode with the `--data` argument and specify your YAML config file, and also include the `--task test` argument.

This will output a performance summary for the model on your test data. For further reference on how to structure your data YAML file and the arguments you can use with the training and validation scripts, please refer to the Ultralytics Documentation.

**Comment by mimansa1907** [comment]
Ok thanks a lot.

**Comment by glenn-jocher** [comment]
@mimansa1907 you're welcome! If you have any more questions or require further assistance as you work with YOLOv8, feel free to reach out. Happy modeling!
